--- a/cmake/find_z3.cmake
+++ b/cmake/find_z3.cmake
@@ -36,15 +36,7 @@
     set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${Z3_INCLUDE_DIRS})
     set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${Z3_LIBRARIES})

-    check_cxx_source_compiles("
-    #include <z3.h>
-    void custom_z3_error_handler(Z3_context ctx, Z3_error_code ec) {
-      ::Z3_string errorMsg = Z3_get_error_msg(ctx, ec);
-    }
-    int main(int argc, char** argv) {
-        return 0;
-    }
-    " HAVE_Z3_GET_ERROR_MSG_NEEDS_CONTEXT)
+    set(HAVE_Z3_GET_ERROR_MSG_NEEDS_CONTEXT 1)
     cmake_pop_check_state()
     if (HAVE_Z3_GET_ERROR_MSG_NEEDS_CONTEXT)
       message(STATUS "Z3_get_error_msg requires context")
